modules/promotion/src/Entity/Promotion.php

```diff
@@ -243,9 +243,23 @@ public function apply(EntityInterface $entity) {
     $entity_type_id = $entity->getEntityTypeId();
     // @todo should whatever invokes this method be providing the context?
     $context = new Context(new ContextDefinition('entity:' . $entity_type_id), $entity);
+
+    // Execute each plugin, this is an AND operation.
+    // @todo support OR operations.
+    /** @var \Drupal\commerce\Plugin\Field\FieldType\PluginItem $item */
+    foreach ($this->get('conditions') as $item) {
```

Comment:
Hmm yah, there should be an applies stand alone, lots of use cases for that. I think apply should include it, but you're right, it is only for preventing someone from doing something stupid.